ephemeris/run_data_managers.py

```diff
@@ -48,32 +127,38 @@ def run_dm(args):
         gi = GalaxyInstance(url=url, email=args.user, password=args.password)
     # should test valid connection
     # The following should throw a ConnectionError when invalid API key or password
-    genomes = gi.genomes.get_genomes()
-    log.info('Number of installed genomes: %s' % str(len(genomes)))
+    genomes = gi.genomes.get_genomes()  # Does not get genomes but preconfigured dbkeys
+    log.info('Number of possible dbkeys: %s' % str(len(genomes)))
+
+    tool_data_client = ToolDataClient(gi)
 
     conf = yaml.load(open(args.config))
+    genomes = conf.get('genomes', '')
     for dm in conf.get('data_managers'):
-        for item in dm.get('items', ['']):
+        items = parse_items(dm.get('items', ['']), genomes)
+        for item in items:
             dm_id = dm['id']
             params = dm['params']
-            log.info('Running DM: %s' % dm_id)
             inputs = dict()
             # Iterate over all parameters, replace occurences of {{item}} with the current processing item
             # and create the tool_inputs dict for running the data manager job
             for param in params:
-                key, value = param.items()[0]
-                value = re.sub(r'{{\s*item\s*}}', item, value, flags=re.IGNORECASE)
+                key, value = list(param.items())[0]
+                value_template = Template(value)
+                value = value_template.render(item=item)
                 inputs.update({key: value})
 
-            # run the DM-job
-            job = gi.tools.run_tool(history_id=None, tool_id=dm_id, tool_inputs=inputs)
-            wait(gi, job)
-            log.info('Reloading data managers table.')
-            for data_table in dm.get('data_table_reload', []):
-                # reload two times
-                for i in range(2):
-                    gi.make_get_request(urljoin(gi.url, 'api/tool_data/%s/reload' % data_table))
-                    time.sleep(5)
+            data_tables = dm.get('data_table_reload', [])
+            # Only run if not run before.
+            if input_entries_exist_in_data_tables(tool_data_client, data_tables, inputs) and not args.overwrite:
+                log.info('%s already run for %s' % (dm_id, inputs))
+            else:
+                log.info('Running DM: "%s" with parameters: %s' % (dm_id, inputs))
+                # run the DM-job
+                job = gi.tools.run_tool(history_id=None, tool_id=dm_id, tool_inputs=inputs)
+                wait(gi, job)
+                log.info('Wait 5 seconds so data tables can be reloaded.')
+                time.sleep(5)
```

Comment:
This isn't necessary, the integration test I've written for galaxy doesn't need a timeout at all, but if you want to be on the safe side you can set this to 1 second.